src/modules/account/RadixAccount.ts

```
@@ -183,15 +220,97 @@ export default class RadixAccount {
+                               ${txUnique}, error: ${error}
+                               `)
+                    }))
+                    .pipe(map((tu: RadixTransactionUpdate) => tu.transaction))
+                    .toPromise()
+            },
+            ).catch(errorFromFaucet => {
+                const errorMsg = `Faucet server is down? ${errorFromFaucet}`
+                logger.error(errorMsg)
+                throw new Error(errorMsg)
+            })
+    }
+
+    public async requestRadsForDevelopmentFromFaucetServiceWithoutBalanceUpdate(
+        maxNumberOfRetries: number = 10,
+        killNodeServerOfFailure: boolean = true,
+    ): Promise<string> {
+
+        for (let attempt = 0; attempt < maxNumberOfRetries; attempt++) {
+            const idOfTxFromFaucet = await RadixAccount.getTokensFromFaucetURL(this.address)
+            if (idOfTxFromFaucet !== undefined) {
+                return idOfTxFromFaucet
+            }
+            logger.error(`Sleeping for ${this.sleepAmount}ms after having failed to get tokens from faucet`)
+            await sleep(this.sleepAmount)
+            this.sleepAmount += sleepAmountIncrease
+            logger.error(`Failed to get tokens from faucet, retrying now... attempt: ${attempt}`)
+        }
+
+        if (killNodeServerOfFailure) {
+            logger.error(`☠️🚨 killing process since we failed to get response from faucet`)
+            process.exit(1)
+        } else {
+            const errorMessage = `Failed to get tokens from faucet`
+            logger.error(errorMessage)
+            throw new Error(errorMessage)
+        }
+    }
+
+    private static async getTokensFromFaucetURL(
+        radixAddress: RadixAddress,
+    ): Promise<string | undefined> {
+        const faucetBaseURL = 'localhost:8079'
+        const faucetURL = `http://${faucetBaseURL}/api/v1/getTokens/${radixAddress.toString()}`
+        return axios.get(faucetURL).then((response) => {
```

`async/await` instead of `.then`

---

need to suppress error, but I think this will work?

```javascript

    private static async getTokensFromFaucetURL(
        radixAddress: RadixAddress,
    ): Promise<string | undefined> {
        const faucetBaseURL = 'localhost:8079'
        const faucetURL = `http://${faucetBaseURL}/api/v1/getTokens/${radixAddress.toString()}`
        const response = await axios.get(faucetURL)
            .catch((e) => {
                logger.error(`🤹🏻‍♂️ Suppressing error: ${e}`)
                return undefined
            })

        if (response.data) {
            const txString = response.data
            if (typeof txString === 'string' && txString.length > 0) {
                return txString
            }
        }

        return undefined

    }
```